app/commands/process_raw_event.rb

```diff
@@ -0,0 +1,127 @@
+class ProcessRawEvent
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(raw_event_id)
+    @raw_event_id = raw_event_id
+  end
+
+  def call
+    raw_event = RawEvent.find_by(id: @raw_event_id)
+    return unless raw_event
+
+    process_raw_event(raw_event)
+  end
+
+  def process_raw_event(raw_event)
+    application = raw_event.application
+
+    parsed_data = parse_raw_data(raw_event)
+
+    meta = parsed_data[:meta]
+    details = parsed_data[:details]
+    details = add_transactions_to_details(details)
+    scenario_hash = generate_scenario_hash(details)
+
+    event_source = find_or_create_event_source(application: application, meta: meta)
+    scenario = find_or_create_scenario(event_source: event_source, hash: scenario_hash)
+    raw_event.scenario = scenario
+    raw_event.save
+
+    events = details.each_with_index.map do |row, index|
+      Event.new(event_hash(scenario: scenario, details: row, meta: meta, index: index))
+    end
+
+    StoreProcessedRequests.call [Request.new(request_hash(scenario: scenario, raw_event: raw_event, meta: meta))]
+    StoreProcessedEvents.call events
+  end
+
+  def parse_raw_data(raw_event)
+    data = JSON.parse(raw_event.data)
+    Parser.new(data).parse
+  end
+
+  def find_or_create_event_source(application:, meta:)
+    application.event_sources.find_or_create_by(source: meta[:controller], endpoint: meta[:action])
+  end
+
+  def find_or_create_scenario(event_source:, hash:)
+    event_source.scenarios.find_or_create_by(events_hash: hash)
+  end
+
+  def request_hash(scenario:, raw_event:, meta:)
+    event_source = scenario.event_source
+
+    {
+      timestamp: Time.at(meta[:start]).to_s(:iso8601),
```

Comment:
Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.at.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.